modules/renter/hostdb/persist_test.go

```diff
@@ -0,0 +1,66 @@
+package hostdb
+
+import (
+	"os"
+	"testing"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// memPersist implements the persister interface in-memory.
+type memPersist hdbPersist
+
+func (m *memPersist) save(data hdbPersist) error { *m = memPersist(data); return nil }
+func (m memPersist) load(data *hdbPersist) error { *data = hdbPersist(m); return nil }
+
+// TestSaveLoad tests that the hostdb can save and load itself.
+func TestSaveLoad(t *testing.T) {
+	// create hostdb with mocked persist dependency
+	hdb := bareHostDB()
+	hdb.persist = new(memPersist)
+
+	// add some fake contracts
+	hdb.contracts = map[types.FileContractID]hostContract{
+		{0}: {IP: "foo"},
+		{1}: {IP: "bar"},
+		{2}: {IP: "baz"},
+	}
```

Comment:
same thing here I think, should be filling out the hostdb via function calls and not direct memory manipulation. The reason is to make it easy to extend. If you start storing metadata on the stuff that gets inserted, or if some extra checking needs to be added or something, or for some reason you change the architecture so multiple fields need to be updated instead of just one, then you can make that change through the function call and not break any of the tests.